### PR TITLE
[Python Wheels] Add more apps

### PR DESCRIPTION
**Pull request by HubertBalcerzak**
This PR adds more applications to be built into python wheels.

Added apps:
* ConvectionDiffusionApplication
* DamApplication
* EigenSolversApplication
* ExternalSolversApplication
* FSIApplication
* ParticleMechanicsApplication
* PoroMechanicsApplication
* SwimmingDEMApplication

Additionally, it is now possible to use cotire when generating windows wheels by providing `-cotire=ON` to the docker run command

### CODE REVIEW

**Review by roigcarlo — DISMISSED** [review]
Great!

**Review by philbucher — COMMENTED** [review]
Can you please wait for my review?
I have some minor comments

File: scripts/wheels/linux/configure.sh [hunk 1]
```diff
@@ -40,4 +48,5 @@ ${CMAKE} -H"${KRATOS_SOURCE}" -B"${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}" \
 -DBOOST_ROOT="/workspace/boost/boost_1_71_0"                           \
 -DLAPACK_LIBRARIES="/usr/lib64/liblapack.so.3"                         \
 -DBLAS_LIBRARIES="/usr/lib64/libblas.so.3"                             \
+-DEIGEN_ROOT="${EIGEN}"                                                \
```
Comment (philbucher) [inline scripts/wheels/linux/configure.sh]:
```suggestion

```

no longer needed

File: scripts/wheels/windows/configure.bat [hunk 2]
```diff
@@ -25,9 +35,9 @@ del /F /Q "%KRATOS_BUILD%\%KRATOS_BUILD_TYPE%\CMakeFiles"
 %cmake% -G"Visual Studio 16 2019" -H"%KRATOS_SOURCE%" -B"%KRATOS_BUILD%\%KRATOS_BUILD_TYPE%"        ^
 -DINCLUDE_FEAST=OFF                                                                                 ^
 -DINSTALL_EMBEDDED_PYTHON=OFF                                                                       ^
--DLAPACK_LIBRARIES=%LAPACK%                                                                         ^
--DBLAS_LIBRARIES=%BLAS%                                                                             ^
--DINSTALL_RUNKRATOS=OFF
+-DEIGEN_ROOT=%EIGEN%                                                                                ^
```
Comment (philbucher) [inline scripts/wheels/windows/configure.bat]:
```suggestion

```

no longer needed

**Review by philbucher — COMMENTED** [review]
minor comments related to Eigen

**Review by philbucher — APPROVED** [review]
looks good to me
thamks @HubertBalcerzak 

@roigcarlo I guess you don't have more comments

**Pull request merged**